src/chapters/thymeleaf-views/studio.rst

```diff
@@ -0,0 +1,88 @@
+
+As always, read through the whole page before starting to code.
+
+Setup
+-----
+
+Fork and clone the appropriate `repository <https://github.com/LaunchCodeEducation/spa-day-starter-code>`_. Check out the project via Version Control in IntelliJ.
+
+When you open the project and run it using ``bootRun``, go to ``localhost:8080`` and you should see a form!
+
+.. figure:: figures/startingform.png
+   :alt: Image showing an empty form with three fields for name, skin type, and whether the person wants a pedicure or manicure.
+
+The Customer Profile
+--------------------
+
+In ``controllers``, we have one controller called ``SpaDayController``. Inside ``SpaDayController``, we have three methods.
+
+#. The ``customerForm()`` method, which looks very similar to what we did in the last lesson.
+#. The ``checkSkinType()`` method. The owners gave us this method to help us figure out which facial treatments are appropriate for which skin type.
+#. The ``spaMenu()`` method, which we will use to bring in our Thymeleaf template, ``menu.html``.
+
+In ``templates``, we have a Thymeleaf template called ``menu.html``.
+Inside ``menu.html``, there are two main divs in the body.
+Let's focus on the div with the id, ``clientProfile``.
+
+#. Add an ``h2`` that says "My profile".
+#. Add a ``p`` tag and use ``th:text`` to bring in the value of ``name``.
+#. Add a ``p`` tag and use ``th:text`` to bring in the value of ``skintype``.
+#. Add a ``p`` tag and use ``th:text`` to bring in the value of ``manipedi``.
+
+Run the application and head to ``localhost:8080`` to see the results! When we fill out the form, we should see a new page with the client profile at the top!
+
+List All Appropriate Facial Treatments
+--------------------------------------
+
+Luckily for us, the spa owners gave us the ``checkSkinType()`` method in our ``SpaDayController``. Also, our teammates already set up code in our ``spaMenu()`` method to fill an ``ArrayList<String>`` with facial treatments that would work for the user's skin type.
+Now, we just need to use Thymeleaf to display the appropriate facial treatments (stored in the ArrayList, ``appropriateFacials``)!
+
+Let's head back to ``menu.html`` and checkout the empty div with the id, ``servicesMenu``.
+
+Add a table and iteratively (using our ``th:each`` and ``th:block`` combo) add rows for the values in ``appropriateFacials``.
+
+Mani or Pedi?
+-------------
+
+One other thing the spa owners want to be cautious of is their treatment descriptions.
+Because the descriptions rarely change and are going to be used in multiple places on the site, the owners have written up the descriptions as Thymeleaf fragments.
+
+Checkout the file, ``fragments.html``, in the ``templates`` directory.
+The owners have already written up the descriptions for their manicure and pedicure in separate ``p`` tags.
+
+We want to put the description in a ``div`` along with an ``h3`` stating that it is either a manicure or pedicure.
```

Comment:
Does this mean that the students will need to add a new ``div`` section in ``menu.html``?  My interpretation of the instructions is that this is necessary, but I had to read the text 2 or 3 times before I came to this understanding.

Reply:
Yes. I can clarify these instructions.